vignettes/teal-data-reproducibility.Rmd

```diff
@@ -10,34 +12,50 @@ vignette: >
+Reproducibility is a primary function of the `qenv` class, which `teal_data` inherits from.
+Every data modification in a `teal_data` object is performed in an encapsulated environment, separate from the global environment.
 
-It is important to note that the reproducibility of this object is limited only to the data-code relationship. Other aspects such as the reliability of the data source, reproducibility of the R session (including package versions), and creation and use of objects from other environments (e.g. `.GlobalEnvironment`) cannot be verified properly by the `teal_data`.
+It is important to note that the reproducibility of this object is limited only to the data-code relationship.
+Other aspects such as the reliability of the data source, reproducibility of the R session (including package versions), and creation and use of objects from other environments (e.g. `.GlobalEnv`) cannot be verified properly by `teal_data`.
+It is advisable to always begin analysis in a new session and run all code that pertains to the analysis within the `teal_data` object.
 
-## Verification status
+## Verification
+
+### Verification status
+
+Every `teal_data` object has a _verification status_, which is a statement of whether the contents of the `env` can be reproduced by `code`.
+From this perspective, `teal_data` objects that are instantiated empty are _verified_ but ones instantiated with data and code are _unverified_ because the code need not be reproducible.
+Obviously, `teal_data` objects instantiated with data only are _unverified_ as well.
+
+When evaluating code in a `teal_data` object, the code that is stored is the same as the code that is executed, so it is reproducible by definition.
+Therefore, evaluating code in a `teal_data` object _does not_ change its verification status.
 
-`teal_data` objects that are instantiated empty are created as _verified_. Objects can be modified only through `eval_code()` and `within()` functions to ensure that the code is saved in the object and always evaluated in the correct environment. Evaluating code in a `teal_data` object _does not_ change its verification status.
+The verification status is always printed when inspecting a `teal_data` object.
+Also, when retrieving code, unverified objects add a warning to the code stating that it has not passed verification.
 
 ```{r, message=FALSE, error=TRUE}
 library(teal.data)
 
-my_data <- teal_data()
-my_data <- within(my_data, data <- data.frame(x = 11:20))
-my_data <- within(my_data, data$id <- seq_len(nrow(data)))
-my_data # is verified
-```
+data_empty <- teal_data()
+data_empty # is verified
+data_empty <- within(data_empty, i <- head(iris))
+data_empty # remains verified
 
-Find out more in the `teal_data()` documentation.
+data_with_data <- teal_data(i = head(iris), code = "i <- tail(iris)")
```

Comment:
Somebody might conclude that data is not verified only because of the wrong code.

```suggestion
data_with_data <- teal_data(i = head(iris), code = "i <- head(iris)")
```

Reply:
Funny, that was what I did initially but then I decided to emphasize that the wrong code can be accepted.

All right, let's keep it simple.